### DISCUSSION TITLE
[@types/react] forwardRef PropsWithoutRef seems to loose all fields(even required one) when there is an indexer in props type

### DISCUSSION THREAD

**Discussion by da1z** [DefinitelyTyped/DefinitelyTyped#56688] [Issues with a @types package]
When there is an indexer in properties type of component with forwardRef it seems like component props are being lost completely, only key and ref are present, even required props are not being highlighted. 

Example:

```
interface MyComponentProps {
    definedProp: string;
    [anyProp: string]: any;
}

const MyComponent = forwardRef((props: MyComponentProps, ref: Ref<any>) => {
    return <div ref={ref} />;
});

export function App() {
    return <MyComponent />;
}
```

Screenshots:
<img width="674" alt="Screen Shot 2021-10-21 at 3 32 01 PM" src="https://user-images.githubusercontent.com/4012422/138366444-1271a0de-6760-41d3-9ebc-271eec7d1575.png">
<img width="1129" alt="Screen Shot 2021-10-21 at 3 32 26 PM" src="https://user-images.githubusercontent.com/4012422/138366760-152f4deb-5e96-4956-93aa-a3fae3af9e9a.png">

Codesandbox:
https://codesandbox.io/s/nifty-water-3dquy?file=/src/App.tsx

[a comment by an automated account is not shown]

**Reply by da1z** [reply]
bump

**Reply by da1z** [reply]
Any update?

**Comment by zbrogz** [comment]
I also ran into this issue. I believe vscode doesn't like the syntax generated from this line: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L814

**Reply by ebrooks42** [reply]
I encountered this issue in a library that my team maintains for others when using creating a component to wrap around react-select, which has an indexer in the exported prop definition by default.

**Comment by dhrumit9** [comment]
This also breaks when used with MUI's `InputBaseComponentProps` as it has an indexer property.
